### ISSUE TITLE
Tests, pre-commit and CI are all broken: 0 tests collected, 23 lint errors, stale 'CI failing' + 'package not found' badges

### ISSUE THREAD

**Issue by LevanBokeria** [alan-turing-institute/heartfm-evals#72]
## Summary

Three separate problems, all currently visible from the repo front page:

1. `pytest` collects **zero tests** and exits with an error.
2. `pre-commit run -a` fails with **20 ruff errors + 3 mypy errors**.
3. The README shows a **"CI failing"** badge and two **"package or version not found"** badges.

None of these are caused by the research code being wrong — the 39 tests all pass once collection is unblocked. They are tooling/config problems that have accumulated since CI was switched off in March.

---

## 1. `pytest` fails at collection — 0 tests run

```
ERROR tests/test_classification_multi_dataset.py - DeprecationWarning: `torch.jit.interface` is deprecated. Please use `torch.compile` instead.
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
```

**Cause.** `pyproject.toml` sets `filterwarnings = ["error"]`, which escalates *every* warning — including third-party ones. `monai.networks.nets.dints` calls `torch.jit.interface` at import time, and recent torch deprecates it. The warning is raised during import of a dependency, so it aborts collection before any test runs.

**Evidence it is only the filter.** With the escalation relaxed:

```
$ pytest -q -W default
39 passed, 1 warning in 10.32s
```

**Proposed fix.** Keep `filterwarnings = ["error"]` (it is valuable for our own code) and add one targeted `ignore` for this specific third-party deprecation, so a regression in our own warnings still fails the suite.

## 2. `pre-commit run -a` fails

The git hook is also **not installed** in a fresh clone (`.git/hooks/pre-commit` is missing unless `pre-commit install` is run), which is how this drifted without anyone noticing.

### ruff — 20 errors

| Count | Rule | What |
| --- | --- | --- |
| 7 | `EXE001` | Shebang present but file not executable — every `scripts/**` driver except `run_segmentation.py`, which *is* `100755`. Inconsistent file modes. |
| 4 | `F401` | Unused imports in `src/`: `torchvision.transforms as T`, `IMAGENET_MEAN`, `IMAGENET_STD` in `data.py`; `set_seed` in `finetune_classification.py` |
| 2 | `F401` | Redundant function-local `import torch` in `decoders.py` (lines 220, 374) — `torch` is already imported at module level |
| 4 | `UP038` | `isinstance(x, (A, B))` → `isinstance(x, A | B)` |
| 1 | `E721` | `type(...) == type(...)` comparison in `compare_classification_results.py` |
| 1 | `ARG001` | Unused `seed` argument in `finetune_sweep_and_train` |
| 1 | `F841` | Unused local `n_samples` in `tests/test_classification_multi_dataset.py` |

Note `F401`/`F841` are in the `unfixable` list, so `ruff --fix` will not touch them — they need reviewing by hand.

### mypy — 3 errors

```
src/heartfm_evals/classification_probe.py:598: error: Item "None" of "Any | None" has no attribute "numpy"  [union-attr]
src/heartfm_evals/classification_probe.py:599: error: Item "None" of "Any | None" has no attribute "numpy"  [union-attr]
src/heartfm_evals/finetune_classification.py:266: error: Argument 1 to "len" has incompatible type "Any | None"; expected "Sized"  [arg-type]
```

All three are the same pattern: `use_val_split = val_features is not None and val_labels is not None` stores the check in a variable, so mypy cannot narrow `Optional` inside `if use_val_split:`. Fix by narrowing in the `if` itself — no behaviour change.

### Related, deliberately *not* fixed here

`set_seed(...)` is imported in three places but every call site is commented out (`0b5d350`, "disabled calling set_seed()"), which is what makes the `set_seed` import unused and the `seed` parameter unused. This looks intentional and related to #59 (*Randomness in fine-tuning classification*), and AGENTS.md warns that seeding is load-bearing for published numbers. **Proposal: silence the lint with an explanatory `noqa` and leave the semantics alone**, tracking the actual re-enable decision in #59.

### "CI failing"

`.github/workflows/` was **deleted** in `78ee0ff` ("removing gh cd and ci for now", 2026-03-27) — the repo has had no workflows since. The badge URL still resolves, and GitHub serves the *last recorded* run for the `CI` workflow, which failed on 2026-03-27. So the badge is permanently pinned to a five-month-old failure.

The old workflows were broken anyway:

- **`ci.yml`** tested `python-version: ["3.10", "3.12"]`, but `requires-python = ">=3.11,<3.14"` — the 3.10 legs could never install the package. (Origin: `.copier-answers.yml` still says `min_python_version: '3.10'`.)
- **`cd.yml`** declared `needs: [pre-commit]` but defined no `pre-commit` job — an invalid workflow. This is visible in `gh run list` as `cd.yml` runs completing in **0s** with `failure`.

**Proposed fix.** Restore a single `ci.yml` that actually matches this project: uv-based (the repo is uv-only per AGENTS.md), Python 3.11 + 3.13, `pre-commit` job + `pytest` job. Note `uv.lock` is gitignored, so CI must resolve dependencies fresh rather than `--frozen`.

### "package or version not found" ×2

Those are the two PyPI badges (`pypi/v` and `pypi/pyversions`). `heartfm-evals` has never been published:

```
$ curl -s -o /dev/null -w "%{http_code}" https://pypi.org/pypi/heartfm-evals/json
404
```

so shields.io renders the "package or version not found" placeholder.

It also **cannot** be published as-is: two dependencies are direct git URLs (`CineMA @ git+...`, `dinov3 @ git+...`), which PyPI rejects. And this is a research benchmark repo, not a distributed library — publishing was never the intent, which is presumably why `cd.yml` was pointed at test.pypi and then deleted.

**Proposed fix.** Drop the two PyPI badges and their link definitions. Keep the Actions badge (fixed by restoring CI).

## 4. Incidental `pyproject.toml` bugs found on the way

- `[tool.coverage]` has `port.exclude_lines` — a typo for `report.exclude_lines`. The whole exclusion list (`pragma: no cover`, `...`, `if typing.TYPE_CHECKING:`) is silently inert.
- `[tool.hatch.metadata] allow-direct-references = true` is vestigial: the build backend is `setuptools`, so the section does nothing (setuptools permits direct references anyway).

## Out of scope / follow-ups

- Pre-commit hook revs are from mid-2024 (`ruff` v0.5.5, `pre-commit-hooks` v4.6.0, `mypy` v1.11.1). Bumping them will surface a new batch of lint errors, so it deserves its own PR rather than being bundled with a CI restore.
- Re-enabling `set_seed()` — see #59.
- `.copier-answers.yml` `min_python_version: '3.10'` disagrees with `requires-python`; only matters on the next `copier update`.

**Comment by LevanBokeria** [comment]
Correction to the ruff tally in the table above — I miscounted while triaging. The 20 errors break down as: `EXE001` ×6 (not 7), `F401` ×8 (3 in `data.py`, 2 redundant function-local `import torch` in `decoders.py`, 3 `set_seed`), `UP038` ×3 (not 4), `E721` ×1, `ARG001` ×1, `F841` ×1. The diagnosis and the fixes in #73 are unaffected.

**Comment by LevanBokeria** [comment]
Fixed in #73 — **CI is now green** ([run 34463358775](https://github.com/alan-turing-institute/heartfm-evals/actions/runs/34463358775)): lint clean, 39 tests passing on both Python 3.11 and 3.13.

Restoring CI turned up a fourth problem that isn't visible from any existing checkout, and it's the most serious one:

**`alan-turing-institute/CineMA` has been blocked by GitHub since 2026-05-04.**

```console
$ gh api repos/alan-turing-institute/CineMA
{"message":"Repository access blocked",
 "block":{"reason":"private_information","created_at":"2026-05-04T17:48:52Z"}}
```

It is not private — it is blocked, and cannot be cloned by anyone, org members included. Since `CineMA @ git+https://github.com/alan-turing-institute/CineMA` is a hard dependency, **`uv sync` has failed for four months**: every fresh clone, new machine and CI run. Our local environments only work because `cinema` was installed in April, before the block.

#73 routes around it by repointing to the upstream fork parent `mathpluscode/CineMA`, which is public and contains the identical commit that is currently installed (`6d74133`) — so the code is unchanged, only its source. Both git dependencies are now pinned to the commits the working environment was built from, since they previously tracked an unpinned HEAD.

@lbokeria the block reason `private_information` is GitHub's category for a takedown over sensitive data in a repository. **That is worth chasing with the org and GitHub support separately from this PR** — #73 only works around the symptom.

Also, one correction to my item 1 above: the `torch.jit.interface` warning is a `DeprecationWarning` on the torch version in our local `.venv` but a `FutureWarning` on the newer torch CI resolves, so the filter has to match by message rather than category. Which incidentally means `torch` is unpinned and CI runs a different version than local dev.
